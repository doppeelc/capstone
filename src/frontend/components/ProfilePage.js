import React, { useState, useEffect, useContext } from "react";
import UserContext from "./UserContext";
import "../styles/ProfilePage.css";
import UserApi from "../api";
import PostFeed from "./PostFeed";


function ProfilePage() {

    const { currentUser } = useContext(UserContext);

    const INIT_STATE = {
                username:currentUser.data.username,
                displayName:currentUser.data.displayName,
                email:currentUser.data.email,
                password:""
            };

    const [formData, setFormData] = useState(INIT_STATE);

    function handleChange(e) {
        const {name, value} = e.target;
        setFormData(formData => ({
            ...formData,
            [name]:value
        }));
    }

    function handleSubmit(e) {
        e.preventDefault();
        UserApi.updateUser(formData);
        setFormData(formData => ({
            ...formData,
            password:""
        }));
    }


    return (
        <div id="ProfilePage">
            <h1>Profile</h1>
            <form id="ProfilePage-form" onSubmit={handleSubmit}>
                <label id="lbl" htmlFor="username">Username</label>
                <input
                    id="inpt"
                    name="username"
                    placeholder="Username"
                    value={formData.username}
                    onChange={handleChange}
                    readOnly />
                
                <label id="lbl" htmlFor="displayName">Display Name</label>
                <input
                    id="inpt"
                    name="displayName"
                    placeholder="Display Name"
                    value={formData.displayName}
                    onChange={handleChange} />
                
                <label id="lbl" htmlFor="email">Email</label>
                <input
                    id="inpt"
                    name="email"
                    placeholder="Email"
                    value={formData.email}
                    onChange={handleChange} />
                    
                <label id="lbl" htmlFor="password">Password</label>
                <input
                    id="inpt"
                    name="password"
                    placeholder="Password"
                    value={formData.password}
                    onChange={handleChange}
                    />
                <button>Update Info</button>
            </form>
            <div className="ProfilePage-posts">
                <PostFeed username={currentUser.data.username} />
            </div>
        </div>
    );
}

export default ProfilePage;