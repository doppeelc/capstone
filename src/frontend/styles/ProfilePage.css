#lbl {
    display: table;
}

#inpt {
    display: table;
    margin-bottom: 10px;
}

#ProfilePage {
    margin: auto;
    text-align: center;
}

#ProfilePage-form {
    width: fit-content;
    margin: auto;
}

#ProfilePage-deleteButton {
    margin: 25px;
    background-color: darkred;
    color: white;
    text-align: center;
}

.ProfilePage-error {
    text-align: center;
    height: 50px;
    margin-bottom: 40px;
}

.ProfilePage-error-text {
    font-size: 20px;
    font-weight: bolder;
    background-color: lightcoral;
    padding: 20px;
    margin-top: 0px;
}