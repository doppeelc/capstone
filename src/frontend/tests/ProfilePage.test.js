import { render } from '@testing-library/react';
import ProfilePage from '../components/ProfilePage';
import UserContext from '../components/UserContext';
import { currentUser } from './__testCommon';
import { BrowserRouter } from 'react-router-dom/cjs/react-router-dom.min';

test('renders without crashing', () => {
    render(
        <BrowserRouter>
            <UserContext.Provider value={{currentUser:currentUser}}>
                <ProfilePage />
            </UserContext.Provider>
        </BrowserRouter>
    );
});

test('matches the snapshot', () => {
    const { asFragment } = render(
        <BrowserRouter>
            <UserContext.Provider value={{currentUser:currentUser}}>
                <ProfilePage />
            </UserContext.Provider>
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

test('renders the form', () => {
    const { getByPlaceholderText } = render(
        <BrowserRouter>
            <UserContext.Provider value={{currentUser:currentUser}}>
                <ProfilePage />
            </UserContext.Provider>
        </BrowserRouter>
    );
    expect(getByPlaceholderText("Username")).toBeInTheDocument();
    expect(getByPlaceholderText("Password")).toBeInTheDocument();
    expect(getByPlaceholderText("Email")).toBeInTheDocument();
    expect(getByPlaceholderText("Display Name")).toBeInTheDocument();
});






