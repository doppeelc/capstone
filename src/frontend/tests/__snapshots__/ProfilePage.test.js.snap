// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    id="ProfilePage"
  >
    <h1>
      Profile
    </h1>
    <form
      id="ProfilePage-form"
    >
      <label
        for="username"
        id="lbl"
      >
        Username
      </label>
      <input
        id="inpt"
        name="username"
        placeholder="Username"
        readonly=""
        value="testuser"
      />
      <label
        for="displayName"
        id="lbl"
      >
        Display Name
      </label>
      <input
        id="inpt"
        name="displayName"
        placeholder="Display Name"
        value="Test User"
      />
      <label
        for="email"
        id="lbl"
      >
        Email
      </label>
      <input
        id="inpt"
        name="email"
        placeholder="Email"
        value="user@test.com"
      />
      <label
        for="password"
        id="lbl"
      >
        Password
      </label>
      <input
        id="inpt"
        name="password"
        placeholder="Password"
        value=""
      />
      <button>
        Update Info
      </button>
    </form>
    <button
      id="ProfilePage-deleteButton"
    >
      Delete Account
    </button>
    <div
      class="ProfilePage-posts"
    >
      <p>
        Loading...
      </p>
    </div>
  </div>
</DocumentFragment>
`;
